server.js

```diff
@@ -31,42 +38,44 @@ app.get('/books', async (request, response) => {
   response.send(books);
 });
 
-app.listen(PORT, () => console.log(`listening on ${PORT}`));
-
-
-
-
-// app.get('/test', (request, response) => {
-//   response.send('test request received');
-// });
-
-
-// const BookModel = mongoose.model('book-collection', bookSchema);
-
-// const book1 = new BookModel(
-//   'Being and Nothingness',
-//   'An Essay on Phenomenological Ontology',
-//   'read',
-//   'bookreader1@gmail.com'
-// );
+app.post('/books', async (request, response) => {
+  const newBook = await Book.create({
+    ...request.body,
+    email: request.query.email,
+  });
+  response.send(newBook);
+});
 
-// const book2 = new BookModel(
-//   'Ham on Rye',
-//   'Auto-biography',
-//   'unread',
-//   'charles@gmail.com'
-// );
+app.delete('/books/:id', async (request, response) => {
+  // if email sent on query, then...
+  if (request.query.email) {
+    // get the book that matches id AND email
+    const foundBook = await Book.findOne({
+      _id: request.params.id,
+      email: request.query.email,
+    });
+
+    // only delete if found
+    console.log({ foundBook });
+
+    // const deleteResult = await Book.findByIdAndDelete(request.params.id);
+
+    if (foundBook) {
+      const deleteResult = await Book.findByIdAndDelete(request.params.id);
+      console.log({ deleteResult });
+      response.status(204).send('success');
+    } else {
+      response.status(400).send('You cannot delete that book'); // correct status code??
+    }
+  } else {
+    response.status(403).send('You must be logged in');
+  }
```

Comment:
You guys are making good progress so nice work!

Feedback here:

- You may want to consider adding in try catch blocks
- Inside `try` you can `await Book.findByIdAndDelete(request.params.id) and res.status(200).send('Success');
- Inside `catch` you will catch errors, so here you will catch the error and res.status(500).('Internal server error error message etc.')